src/overlays/actors/ovl_En_Zf/z_en_zf.c

```diff
@@ -82,115 +151,2291 @@ static ColliderQuadInit D_80B4A204 = {
+        } else {
+            this->actor.speedXZ = 0.0f;
+            this->hopAnimIndex = 0;
+            EnZf_SetupApproachPlayer(this, globalCtx);
+        }
+    }
+    if ((globalCtx->gameplayFrames & 0x5F) == 0) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_RIZA_CRY);
+    }
+
+    if ((this->actor.params == ENZF_TYPE_DINOLFOS) && (this->actor.bgCheckFlags & 3)) {
+        if (EnZf_MayAttack(globalCtx, this)) {
+            EnZf_SetupSlash(this);
+        } else {
+            func_80B483E4(this, globalCtx);
+        }
+    }
+}
+
+void func_80B4604C(EnZf* this) {
+    Animation_MorphToLoop(&this->skelAnime, &gZfWalkingAnim, -4.0f);
+    this->action = ENZF_ACTION_6;
+    EnZf_SetupAction(this, func_80B46098);
+}
+
+void func_80B46098(EnZf* this, GlobalContext* globalCtx) {
+    s32 pad;
+    f32 phi_f2;
+    Player* player = GET_PLAYER(globalCtx);
+    s16 temp_v0;
+    s16 phi_v1;
+
+    if (!EnZf_DodgeRangedEngaging(globalCtx, this)) {
+        if ((this->actor.params != ENZF_TYPE_DINOLFOS) || !EnZf_ChooseAction(globalCtx, this)) {
+            temp_v0 = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+
+            if (temp_v0 > 0) {
+                phi_v1 = (temp_v0 * 0.25f) + 2000.0f;
+            } else {
+                phi_v1 = (temp_v0 * 0.25f) - 2000.0f;
+            }
+
+            this->actor.shape.rot.y += phi_v1;
+            this->actor.world.rot.y = this->actor.shape.rot.y;
+
+            if (temp_v0 > 0) {
+                phi_f2 = phi_v1 * 1.5f;
+                if (phi_f2 > 2.0f) {
+                    phi_f2 = 2.0f;
+                }
+            } else {
+                phi_f2 = phi_v1 * 1.5f;
+                if (phi_f2 < -2.0f) {
+                    phi_f2 = -2.0f;
+                }
+            }
+
+            this->skelAnime.playSpeed = -phi_f2;
+            SkelAnime_Update(&this->skelAnime);
+
+            // miniboss
+            if (this->actor.params >= ENZF_TYPE_LIZALFOS_MINIBOSS_A) {
+                this->curPlatform = EnZf_FindPlatform(&this->actor.world.pos, this->curPlatform);
+                if (this->curPlatform != EnZf_FindPlatform(&player->actor.world.pos, -1)) {
+                    EnZf_SetupApproachPlayer(this, globalCtx);
+                    return;
+                }
+            }
+
+            if (Actor_IsFacingPlayer(&this->actor, 30 * 0x10000 / 360)) {
+                if (Rand_ZeroOne() > 0.8f) {
+                    func_80B462E4(this, globalCtx);
+                } else {
+                    EnZf_SetupApproachPlayer(this, globalCtx);
+                }
+            }
+            if ((globalCtx->gameplayFrames & 0x5F) == 0) {
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_RIZA_CRY);
+            }
+        }
+    }
+}
+
+// Conditional setup function
+void func_80B462E4(EnZf* this, GlobalContext* globalCtx) {
+    // miniboss
+    if ((this->actor.params < ENZF_TYPE_LIZALFOS_MINIBOSS_A) ||
```

Comment:
```suggestion
    if ((this->actor.params < ENZF_TYPE_LIZALFOS_MINIBOSS_A) || /* not miniboss */
```

Reply:
This pattern is used so often that I'm thinking it would be better to have a define for `IS_MINIBOSS`, but I suspect people will think that's less clear? I'd argue the opposite, actually, which is why I've added all these comments.

Reply:
yeah I thought about it, I would be ok with it. but I also don't think we can reach consensus about using a define, let's wait for more opinions